patches/server/0994-Add-drops-to-shear-events.patch

```diff
@@ -0,0 +1,275 @@
+                 this.discard(); // CraftBukkit - from above
+                 // CraftBukkit end
+ 
++                if (drops == null) { // Paper
+                 for (int i = 0; i < 5; ++i) {
+                     // CraftBukkit start
+                     ItemEntity entityitem = new ItemEntity(this.level(), this.getX(), this.getY(1.0D), this.getZ(), new ItemStack(this.getVariant().blockState.getBlock()));
+@@ -211,6 +231,11 @@ public class MushroomCow extends Cow implements Shearable, VariantHolder<Mushroo
+                     this.level().addFreshEntity(entityitem);
+                     // CraftBukkit end
+                 }
++                // Paper start - add drops from shear events
++                } else {
++                    drops.forEach(bukkitItemStack -> this.spawnAtLocation(new ItemEntity(this.level(), this.getX(), this.getY(1.0D), this.getZ(), org.bukkit.craftbukkit.inventory.CraftItemStack.asNMSCopy(bukkitItemStack))));
++                }
++                // Paper end
+             }
+         }
+ 
+diff --git a/src/main/java/net/minecraft/world/entity/animal/Sheep.java b/src/main/java/net/minecraft/world/entity/animal/Sheep.java
+index c0e89262c596fbdd0bb3c3f76baccb17a1bb5fcd..4127f6ae9de20974a2eaebb44e0a2ba6a07c41c4 100644
+--- a/src/main/java/net/minecraft/world/entity/animal/Sheep.java
++++ b/src/main/java/net/minecraft/world/entity/animal/Sheep.java
+@@ -251,11 +251,14 @@ public class Sheep extends Animal implements Shearable {
+         if (itemstack.is(Items.SHEARS)) {
+             if (!this.level().isClientSide && this.readyForShearing()) {
+                 // CraftBukkit start
+-                if (!CraftEventFactory.handlePlayerShearEntityEvent(player, this, itemstack, hand)) {
++                // Paper start
++                org.bukkit.event.player.PlayerShearEntityEvent event = CraftEventFactory.handlePlayerShearEntityEvent(player, this, itemstack, hand, generateDefaultDrops());
++                if (event != null && event.isCancelled()) {
++                    // Paper end
+                     return InteractionResult.PASS;
+                 }
+                 // CraftBukkit end
+-                this.shear(SoundSource.PLAYERS);
++                this.shear(SoundSource.PLAYERS, event == null ? null : event.getDrops()); // Paper - pass event drops
+                 this.gameEvent(GameEvent.SHEAR, player);
+                 itemstack.hurtAndBreak(1, player, (entityhuman1) -> {
+                     entityhuman1.broadcastBreakEvent(hand);
+@@ -271,10 +274,28 @@ public class Sheep extends Animal implements Shearable {
+ 
+     @Override
+     public void shear(SoundSource shearedSoundCategory) {
++        // Paper start
++        this.shear(shearedSoundCategory, null);
++    }
++
++    @Override
++    public java.util.List<ItemStack> generateDefaultDrops() {
++        int count = this.random.nextInt(3);
++        java.util.List<ItemStack> dropEntities = new java.util.ArrayList<>(count);
```

Comment:
The precomputed size should be: 1 + count
or you can increase the count var directly